tests/csm/rest/test_iam_users.py

```diff
@@ -450,12 +673,29 @@ def test_35931(self):
             resp = self.csm_obj.compare_iam_payload_response(resp1, optional_payload)
             self.log.info("Printing response %s", resp)
             assert_utils.assert_true(resp[0], resp[1])
-            self.log.info("Create bucket and perform IO")
+            self.log.info("Step: Verify no bucket present in new account")
             s3_obj = S3TestLib(access_key=resp1.json()["keys"][0]["access_key"],
                                secret_key=resp1.json()["keys"][0]["secret_key"])
+            buckets = s3_obj.bucket_list()[1]
+            assert_utils.assert_false(len(buckets), "buckets found on new IAM user")
+            self.log.info("Step: Verified no bucket present in new account")
+            self.log.info("Create bucket and perform IO")
             self.log.info("Step: Verify create bucket")
-            status, resp = s3_obj.create_bucket(bucket_name)
+            status, resp = s3_obj.create_bucket(self.bucket_name)
```

Comment:
Lines 684 - 708 can be replaced by create_bucket_put_object function from s3_common_test_lib.